testprograms/test_ppc_PWR10.c

```diff
@@ -0,0 +1,66 @@
+#define GNUC_VERSION (__GNUC__*1000 + __GNUC_MAJOR__*10)
+#if (GNUC_VERSION >= 14010) || defined(__clang__)
```

Comment:
I suggest to use macro `__GNUC_PREREQ` from `<features.h>` instead.

Reply:
yeeeew. I copied this from Jeff Walton who wanted Clang support.

It looks like __GNUC_PREREQ will fail for Clang because Clang is rude enough to define bogus values for __GNUC__ and __GNUC_MINOR__ (4.2 as I recall). 

Also I am not sure why/if this pragma is needed.
pragma GCC diagnostic ignored "-Wdeprecated"

Anyway P10 requires GCC >= 10 (AT14) So anything else will fail the configure test.

But I looks like I confused AT14 and GCC 10. So should be the test should be:
GNUC_VERSION >= 10010

So I'll fix that. But the larger question will have to wait.